docs/LoaderApplicationInterface.md

```diff
@@ -0,0 +1,663 @@
+VK_LAYER_KHRONOS_validation layer.
+
+```
+char *instance_validation_layers[] = {
+    "VK_LAYER_KHRONOS_validation"
+};
+const VkApplicationInfo app = {
+    .sType = VK_STRUCTURE_TYPE_APPLICATION_INFO,
+    .pNext = NULL,
+    .pApplicationName = "TEST_APP",
+    .applicationVersion = 0,
+    .pEngineName = "TEST_ENGINE",
+    .engineVersion = 0,
+    .apiVersion = VK_API_VERSION_1_0,
+};
+VkInstanceCreateInfo inst_info = {
+    .sType = VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO,
+    .pNext = NULL,
+    .pApplicationInfo = &app,
+    .enabledLayerCount = 1,
+    .ppEnabledLayerNames = (const char *const *)instance_validation_layers,
+    .enabledExtensionCount = 0,
+    .ppEnabledExtensionNames = NULL,
+};
+err = vkCreateInstance(&inst_info, NULL, &demo->inst);
+```
+
+At `vkCreateInstance` and `vkCreateDevice`, the loader constructs call chains
+that include the application specified (enabled) layers.
+Order is important in the `ppEnabledLayerNames` array; array element 0 is the
+topmost (closest to the application) layer inserted in the chain and the last
+array element is closest to the driver.
+See the [Overall Layer Ordering](#overall-layer-ordering) section for more
+information on layer ordering.
+
+**NOTE:** *Device Layers Are Now Deprecated*
+> `vkCreateDevice` originally was able to select layers in a similar manner to
+`vkCreateInstance`.
+> This lead to the concept of "instance layers" and "device layers".
+> It was decided by Khronos to deprecate the "device layer" functionality and
+> only consider "instance layers".
+> Therefore, `vkCreateDevice` will use the layers specified at
+`vkCreateInstance`.
+> Because of this, the following items have been deprecated:
+> * `VkDeviceCreateInfo` fields:
+>   * `ppEnabledLayerNames`
+>   * `enabledLayerCount`
+> * The `vkEnumerateDeviceLayerProperties` function
+
+
+### Meta-Layers
+
+Meta-layers are layers which really just contain an ordered list of other
+layers.
+This is to allow grouping layers together in a specified order so that they can
+interact properly.
+Originally, this was designed to group together the individual Vulkan
+Validation layers in the proper order to avoid conflicts.
+While the Vulkan Validation layer no longer uses this, VkConfig does use this
+to group layers together based on user's desires.
+More can be found out about this functionality through both the
+[VkConfig documentation](https://github.com/LunarG/VulkanTools/blob/master/vkconfig/README.md)
+and the section later on the [Override Layer](#override-layer).
+
+Meta-layers are detailed more in the
+[Meta-Layers](LoaderLayerInterface.md#meta-layers) section of the
+[LoaderLayerInterface.md](LoaderLayerInterface.md) file in this folder.
+
+
+### Implicit vs Explicit Layers
+
+![Different Types of Layers](./images/loader_layer_order.png)
+
+Explicit layers are layers which are enabled by an application (e.g. with the
+vkCreateInstance function as mentioned previously), or by an environment
+variable.
```

Comment:
Technically, layers enabled with an environment variable are implicit layers. They just are *not* enabled all the time. I agree with the intent of this sentence, but not the specifics. (Because to the loader, Implicit layers means layers with X properties in the json manifest. Layers that get enabled with an env-var are use the rules defined by implicit layers, not explicit.
The following sentences discusses 'certain application environments' which are what env-var enabled layers usually are used for.

Reply:
Good point, really the only thing you can do with explicit layers with environment variables is change their path.  I'll remove the last part of this.

Reply:
Done, please look over new version.

Reply:
`Explicit layers are layers which are enabled by an application (e.g. with the
vkCreateInstance function as mentioned previously).`